routes.js

```diff
@@ -0,0 +1,327 @@
+import express from 'express';
+import path from 'path';
+import Joi from 'joi';
+import { requireJsonContent } from './middleware/index';
+import netflixData from './data/netflix-titles.json';
+
+const router = express.Router();
+const ITEMS_PER_PAGE = 20;
+let remainingPages = null;
+let totalPages = null;
+let movies = netflixData;
+
+// GET routes
+router.get('/', (req, res) => {
+  res.sendFile(path.join(__dirname + '/index.html'));
+});
+
+router.get('/movies', (req, res) => {
+  const queryPage = +req.query.page;
+  const queryYear = +req.query.year;
+  const queryDuration = +req.query.duration;
+  const queryActor = req.query.actor;
+
+  // Local copy of movies
+  let filteredMovies = movies;
+
+  // Query - Year
+  if (queryYear) {
+    filteredMovies = filteredMovies.filter(
+      movie => movie.release_year === queryYear
+    );
+  }
+
+  // Query - Actor
+  if (queryActor) {
+    filteredMovies = filteredMovies.filter(movie => {
+      let actors = movie.cast.split(', ');
+
+      // Check if current movie contains value in queryActor
+      const result = actors.filter(item => {
+        // Create regular expression with dynamic parameter
+        let regexp = new RegExp(
+          '\\b(' + queryActor.toLowerCase() + ')\\b',
+          'gi'
+        );
+
+        // Check for match
+        const match = item.toLowerCase().match(regexp);
+
+        // If actor is present in current movie item is returned
+        if (match) {
+          return item;
+        } else {
+          return;
+        }
+      });
+
+      // Return current movie if actor is found
+      if (result.length > 0) {
+        return movie;
+      }
+    });
+  }
```

Comment:
Try to avoid having too much logic going on in one place. This filter is pretty difficult to follow, with a filter in a filter, and returning the item rather than a boolean - it makes it all a bit difficult to mentally parse. When the logic starts getting thick like this, try to push yourself to break things up a little more and make use of functions to encapsulate and self-document what's going on. 👍